File: 1004-max-consecutive-ones-iii/solution.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int result = 0;
        int left=0; 
        int right =0;
        deque<int>dq;
        while(right<nums.size())
        {
            if(nums[right] == 0)
            {
                dq.push_back(right);
            }

            if(dq.size()>k)
            {
                left = dq.front() + 1;
                dq.pop_front();
            }
            result = max(result,right - left +1);
            right++;
        }
        return result;
    }
};

File: 1004-max-consecutive-ones-iii/solution.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:

        n = len(nums)
        l = 0
        countZero = 0
        result = 0

        for r in range(0,n):
            if nums[r] == 0 :
                countZero +=1
            
            if countZero >k:
                while countZero>k and l<n:
                    if nums[l] == 0:
                        countZero -=1
                    
                    l+=1
            
            result = max(result,r-l+1)

        return result
    


        